### PR TITLE
Support arithmetic expressions and literals in Datalog predicates

### PR DESCRIPTION
**Pull request by shadaj**
Support arithmetic expressions and literals in Datalog predicates

---
Stack created with [Sapling](https://sapling-scm.com). Best reviewed with [ReviewStack](https://reviewstack.dev/hydro-project/hydroflow/pull/540).
* #543
* #542
* #541
* __->__ #540

### CODE REVIEW

**Comment by shadaj** [comment]
@tylerhou should I just integrate your changes? or should we stack

File: hydroflow/tests/datalog_frontend.rs [hunk 1]
```diff
@@ -883,3 +883,29 @@ fn test_less_than_relation() {
...
+fn test_expr_predicate() {
+    let (ints_send, ints) = hydroflow::util::unbounded_channel::<(i64,)>();
+    let (result, mut result_recv) = hydroflow::util::unbounded_channel::<(i64,)>();
+
+    let mut flow = datalog!(
+        r#"
+        .input ints `source_stream(ints)`
+        .output result `for_each(|v| result.send(v).unwrap())`
+
+        result(1) :- ints(a), (a == 0)
```
Comment (tylerhou) [inline hydroflow/tests/datalog_frontend.rs:897]: Maybe a test with non-trivial expressions on both sides? `a - 1 == 1 - 1`?

File: hydroflow_datalog_core/src/grammar.rs [hunk 2]
```diff
@@ -179,16 +179,17 @@ pub mod datalog {
...
+        Neq(#[rust_sitter::leaf(text = "!=")] ()),
     }
 
     #[derive(Debug, Clone)]
     pub struct PredicateExpr {
         #[rust_sitter::leaf(text = "(")]
         _l_brace: (),
 
-        pub left: Spanned<Ident>,
+        pub left: Spanned<ValueExpr>,
```
Comment (tylerhou) [inline hydroflow_datalog_core/src/grammar.rs:190]: How do you feel about renaming ValueExpr to IntExpr?

**Review by tylerhou — APPROVED** [review]

**Comment by tylerhou** [comment]
I'll rebase mine onto yours, seems easier.

File: hydroflow_datalog_core/src/grammar.rs (hunk 2 above)
Comment (shadaj) [reply hydroflow_datalog_core/src/grammar.rs:190]: I like this idea

**Pull request merged**